StyleCop.Analyzers/StyleCop.Analyzers/Helpers/SpecializedTasks.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace StyleCop.Analyzers.Helpers
+{
+    using System.Threading.Tasks;
+
+    internal static class SpecializedTasks
+    {
+        internal static Task CompletedTask { get; } = Task.FromResult(default(VoidResult));
```

Comment:
Good catch. Didn't find any documentation about that except source code:
https://github.com/dotnet/roslyn/blob/918dd32ccdfd901bfd71d88bdb9f7051adfb3d51/src/Features/Core/CodeFixes/CodeFixService.cs#L219

https://github.com/dotnet/roslyn/blob/918dd32ccdfd901bfd71d88bdb9f7051adfb3d51/src/Features/Core/CodeFixes/CodeFixService.cs#L393

https://github.com/dotnet/roslyn/blob/3faf03a3e52c1b46bbc3967060cf6406d4ecc9af/src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/BatchFixAllProvider.cs#L111

in all case they use own cached task if user's implementation of `RegisterCodeFixesAsync` returns `null`.